src/main/java/org/springframework/data/mapping/PropertyPath.java

```diff
@@ -252,6 +281,11 @@ public static PropertyPath from(String source, TypeInformation<?> type) {
 			}
 		}
 
+		if (result == null) {
+			throw new IllegalStateException(
```

Comment:
Is that specific to Eclipse? In IntelliJ, I'm using `Assert.state(result != null, ...)` all over the place... checking for not-null, of course, but I guess that was just a typo :-)